datatypes/src/jvmMain/kotlin/at/asitplus/crypto/datatypes/JcaExtensions.kt

```diff
@@ -75,29 +75,31 @@ fun CryptoPublicKey.Rsa.getPublicKey(): RSAPublicKey =
     ) as RSAPublicKey
 
 @Throws(Throwable::class)
-fun CryptoPublicKey.Ec.Companion.fromJcaKey(publicKey: ECPublicKey): CryptoPublicKey.Ec {
-    val curve = EcCurve.byJcaName(
-        SECNamedCurves.getName(
-            SubjectPublicKeyInfo.getInstance(
-                ASN1Sequence.getInstance(publicKey.encoded)
-            ).algorithm.parameters as ASN1ObjectIdentifier
+fun CryptoPublicKey.Ec.Companion.fromJcaKey(publicKey: ECPublicKey): KmmResult<CryptoPublicKey.Ec> =
+    runCatching {
+        val curve = EcCurve.byJcaName(
+            SECNamedCurves.getName(
+                SubjectPublicKeyInfo.getInstance(
+                    ASN1Sequence.getInstance(publicKey.encoded)
+                ).algorithm.parameters as ASN1ObjectIdentifier
+            )
+        ) ?: throw SerializationException("Unknown Jca name")
+        fromCoordinates(
+            curve,
+            publicKey.w.affineX.toByteArray().ensureSize(curve.coordinateLengthBytes),
+            publicKey.w.affineY.toByteArray().ensureSize(curve.coordinateLengthBytes)
         )
-    ) ?: throw SerializationException("Unknown Jca name")
-    return fromCoordinates(
-        curve,
-        publicKey.w.affineX.toByteArray().ensureSize(curve.coordinateLengthBytes),
-        publicKey.w.affineY.toByteArray().ensureSize(curve.coordinateLengthBytes)
-    )
-}
+    }.wrap()
 
-fun CryptoPublicKey.Rsa.Companion.fromJcaKey(publicKey: RSAPublicKey): CryptoPublicKey.Rsa =
-    CryptoPublicKey.Rsa(publicKey.modulus.toByteArray(), publicKey.publicExponent.toInt())
+fun CryptoPublicKey.Rsa.Companion.fromJcaKey(publicKey: RSAPublicKey): KmmResult<CryptoPublicKey.Rsa> =
+    runCatching { CryptoPublicKey.Rsa(publicKey.modulus.toByteArray(), publicKey.publicExponent.toInt()) }.wrap()
 
+@Throws(Throwable::class)
 fun CryptoPublicKey.Companion.fromJcaKey(publicKey: PublicKey): KmmResult<CryptoPublicKey> =
     runCatching {
         when (publicKey) {
-            is RSAPublicKey -> CryptoPublicKey.Rsa.fromJcaKey(publicKey)
-            is ECPublicKey -> CryptoPublicKey.Ec.fromJcaKey(publicKey)
+            is RSAPublicKey -> CryptoPublicKey.Rsa.fromJcaKey(publicKey).getOrThrow()
+            is ECPublicKey -> CryptoPublicKey.Ec.fromJcaKey(publicKey).getOrThrow()
             else -> throw IllegalArgumentException("Unsupported Key Type")
```

Comment:
don't `getOrThrow()` just to wrap it: better just return it (as is is a KmmResult already) and explicitly return a failure for the `else` branch